src/core/logging/logger.ts

```diff
@@ -0,0 +1,53 @@
+import { BotConfig } from "../../types/core/botConfig";
+import { DiscordLogger } from "./discordLogger";
+import { SlackLogger } from "./slackLogger";
+
+/**
+ *
+ */
+export class Logger {
+	private botConfig: BotConfig;
+	public discordLogger?: DiscordLogger;
+	public slackLogger?: SlackLogger;
+
+	// Codes that are not sent to external sources (discord, slack)
+	private externalExemptCodes: Array<number> = [4];
+
+	/**
+	 *
+	 */
+	constructor(config: BotConfig) {
+		this.botConfig = config;
+
+		if (this.botConfig.discordWebhookUrl) {
+			this.discordLogger = new DiscordLogger(this.botConfig.discordWebhookUrl);
+		}
+
+		if (this.botConfig.slackToken && this.botConfig.slackChannel) {
+			this.slackLogger = new SlackLogger(this.botConfig.slackToken, this.botConfig.slackChannel);
+		}
+	}
+
+	/**
+	 * Sends the `message` to the console and to discord or slack if defined.
+	 * @param message The message to log.
+	 * @param code The code number of the message, -1 if not given.
+	 */
+	public async sendMessage(message: string, code = -1) {
```

Comment:
Consider changing the behavior of the logger in a sense that you can decide to log a message to discord, slack, or both. Something along the line:
sendMessage(m, discord=true, slack=false). That gives you more granular control about what to log where. So for example. you could setup a channel for "wins" and one for general logging.

Reply:
Sure, could do an parameter based on an enum where log types could be defined. I'm not sure the targeting individual accounts is the best for those types though.  Probably better to be more generic around content / event being sent.

I was using the `code` param to determine if a win/fail with `externalExemptCodes` and an option to filter the errors from things getting sent to external accounts (slack, discord, etc.). Basically having console logs being the catch all for all errors.

Reply:
Yeah that could also make sense. Just want to make sure that we build the logger rather flexible than restricting.

Reply:
Added enum and updated logger usage on main loops. Let me know your thoughts. 

In a future iteration we could have a config value to that sets what level of logs should go to the external messaging systems. Some might enjoy having everything, others might just want success, etc.